Pixhawk 6X-RT Add bootloader update guide (#3008)
PX4/PX4-user_guide: en/advanced_config/bootloader_update.md

   In this case you will not be able to install FMUv3 hardware.
   :::

## Debug Probe Bootloader Update

The following steps explain how you can "manually" update the bootloader using a [compatible Debug Probe](../debug/swd_debug.md#debug-probes-for-px4-hardware):

1. Get a binary containing the bootloader (either from dev team or build it yourself).

1. Get a [Debug Probe](../debug/swd_debug.md#debug-probes-for-px4-hardware).
   Connect the probe your PC via USB and setup the `gdbserver`.

1. Go into the directory containing the binary and run the following command in the terminal:

   ```sh

>    arm-none-eabi-gdb px4fmuv5_bl.elf

Reviewer: ```suggestion
   arm-none-eabi-gdb px4fmuv6x_bl.elf
```

Author: This is incorrect so legacy for V5 is 
`px4fmuv5_bl.elf`
Older version I don't even know.
For V6X it's
`px4_fmu-v6x_bootloader.elf`
For V6X-RT it's
`px4_fmu-v6xrt_bootloader.elf`

Reviewer 2: In-tree H7 Booloaders from https://github.com/PX4/PX4-Autopilot are named *._bootloader all the others are from https://github.com/PX4/PX4-Bootloader and end with _bl. 

Using up a term like legacy will not help anyone nor will it be valid on the next bootloaderr.

Reviewer: Thanks both. Fixed by listing the ones above and adding a note about the pattern.

> Using up a term like legacy will not help anyone nor will it be valid on the next bootloaderr.

You could not be more right. Sorry about that.

   ```

   :::note

>    If the hardware is FMUv6X or newer use the in-tree PX4 bootloader.

Reviewer: ```suggestion
   The naming pattern for the bootloader binary is the same for both in-tree and legacy PX4 bootloader.
```

Author: In correct see comment above.

>    ```sh
>    arm-none-eabi-gdb px4fmuv5_bl.elf
>    ```
> 
>    :::note
>    If the hardware is FMUv6X or newer use the in-tree PX4 bootloader.
>    :::

Reviewer: I think this would be better? (please check command)

```suggestion
   ```sh
   arm-none-eabi-gdb px4fmuv6x_bl.elf
   ```

   :::note
   The naming pattern for the bootloader binary is the same for both in-tree and legacy PX4 bootloader.
   :::
```

   In this case you will not be able to install FMUv3 hardware.
   :::

> ## Debug Probe Bootloader Update

Reviewer 2: @mrpollo  Is this correct and OK to change

Reviewer: This is _good_ to change, provided the instructions worth with all the linked probes. I will assume so.

   Reading symbols from px4fmuv5_bl.elf...done.
   ```

1. Find your `<dronecode-probe-id>` by running an `ls` command in the **/dev/serial/by-id** directory.

1. Now connect to the debug probe with the following command:

   ```sh
   tar ext /dev/serial/by-id/<dronecode-probe-id>
   ```

> 1. Power on the Pixhawk with another USB cable and connect the probe to the `FMU-DEBUG` port.

Reviewer: Assumption here is all probes connect to this port.

Author: Yes correct

   :::note

>    To be able to connect a Dronecode probe to the `FMU-DEBUG` port, you may need to remove the case (e.g. on Pixhawk 4 you would do this using a T6 Torx screwdriver).

Reviewer: ```suggestion
   If using a Dronecode probe you may need to remove the case in order to connect to the `FMU-DEBUG` port (e.g. on Pixhawk 4 you would do this using a T6 Torx screwdriver).
```
